Sources/NIOPosix/SocketChannel.swift

```diff
@@ -552,6 +563,19 @@ final class DatagramChannel: BaseSocketChannel<Socket> {
                 // Receiving packet info is only supported for IP
                 throw ChannelError.operationUnsupported
             }
+        case _ as ChannelOptions.Types.DatagramGenericSegmentationOffload:
+            if System.supportsUDPSegmentationOffload {
+                #if os(Linux)
```

Comment:
Instead of putting the compile time checks in here, can we put them into functions in `BSDSocketAPI` with different implementations in the various placse? I'd like us to avoid too much multiplication of `#if os` throughout this file, so if we can split these branches into functions that exist on all platforms but that are only implemented non-trivially on the platforms we care about, that would be very helpful.